### ISSUE TITLE
Tree Style Tab and Simple-tab-groups : All trees are flattened when restoring a stg-backup

### ISSUE THREAD

**Issue by Elaws** [Drive4ik/simple-tab-groups#956]

**Comment by SHHSSH** [comment]
This does seem to be regularly occurring when restoring. I have come to deal with it.

Since I use TST a lot too, I was wondering, have you experienced a lot of errors with STG? Sometimes it seems my settings malfunction or something and I have to totally reinstall the add-on.

**Comment by Elaws** [comment]
STG and TST don't interact well at all, to the point it's not viable. I'm currently looking for another solution.

**Comment by SHHSSH** [comment]
> STG and TST don't interact well at all, to the point it's currently not viable. I'm looking for another solution.

Unfortunately I came to a similar conclusion but couldn't live without TST, so, I ended up looking for other solutions by categorising tabs into folders and on the odd occasion I will either bookmark them, or export all tabs into a text document.

<a href="https://ibb.co/09kN2hc"><img src="https://i.ibb.co/09kN2hc/capture-020-22092022-024441.jpg" alt="capture-020-22092022-024441" border="0"></a>

**Comment by Elaws** [comment]
This is a good idea, I will experiment with that. However folders can't be hotkeyed ? And when you open a folder, is there an option to hide tabs that are in the other folders ?

I discovered Sidebery add-on lately, seems to do everything STG + TST does, but unfortunately it does not automatically restore previous session -_-', which is quite odd. Also, not sure how reliable it is.

Well, quite fed up with my containers & tree structures constantly messed up, so I may experiment with your idea !

**Comment by Elaws** [comment]
(Off-topic : Btw, have you noticed lately that all the icons of tabs disappear when you quit/relaunch Firefox ?)

**Comment by SHHSSH** [comment]
> And when you open a folder, is there an option to hide tabs that are in the other folders ?

Yes, there is an option that collapses other folders' hierarchies when a folder hierarchy is selected. This can actually be a finnickey thing to optimise to your liking, due to it occasionally conflicting with other options of the sort in TST. 

But with the helper addon by the same developer "Lock as collapsed". you can lock folder hierarchies.
You can then in turn add macros to collapse/extend the "tree/folder".

Personally, I then select through the tabs with a GShiftKey+Scroll Left/Right -(Just macroed CtrlTab/ShifTab).

> Well, quite fed up with my containers & tree structures constantly messed up, so I may experiment with your idea !

I've found nothing has come close to TST's features/customisability/stability and I've used darn near almost all _Firefox_ addons relevant to curation of sorts.

> (Off-topic : Btw, have you noticed lately that all the icons of tabs disappear when you quit/relaunch Firefox ?)

Short answer; no, it sounds like a profile or cache corruption, potentially due to past add on changes.

A lot of your problems sound like you could really do with a bit of a profile refresh tbh. If not that, you should really look into how many addons you are actually running, I'm sure you're just trying to find the optimal solution for your workflow but make sure that you're not bloating your browser with excessive _glizy/gimmicky_ "productivity" features that in the long run create more procrastination than performance optimisation.

Check out this website if you intend on revamping your profile - https://ffprofile.com/ - I definitely had a tonne of problems with a excess dev edition usage whilst tweaking/testing things after a while.

Here's a few addons I use primarily, if you care, but even these are often toggled off.

<a href="https://ibb.co/jLqmBQd"><img src="https://i.ibb.co/jLqmBQd/capture-023-22092022-045158.jpg" alt="capture-023-22092022-045158" border="0"></a>

<a href="https://ibb.co/mGDftq7"><img src="https://i.ibb.co/mGDftq7/capture-024-22092022-045204.jpg" alt="capture-024-22092022-045204" border="0"></a>

**Comment by SHHSSH** [comment]
I see you're an Obsidian user.

I've recently just jumped aboard, was in the works for a while.
One of the first things I changed was being able to CtrlTab/ShiftTab through documents in it too.

I also use a customised layout of Tablacus Explorer - [url=https://ibb.co/M2bD8Wh][img]https://i.ibb.co/M2bD8Wh/capture-026-22092022-051919.jpg[/img][/url]
A more productive explorer in my opinion.

Be well friend, I may just ask you for some help in Obsidian one day, there's many things I feel lost with in it.

**Comment by Elaws** [comment]
Thanks for tips about ffprofile and tablacus !

Well, after losing again today all my tree structure and panels, I decided to test Sidebery more extensively. Not sure why it didn't properly restore session yesterday (and I don't have many add-ons), but today everything is working perfectly fine. So far, it's a wonder, and does everything I used in TST+STG. If it's reliable within the next few days/weeks, I'm gonna completely switch.

![0Sidebery](https://user-images.githubusercontent.com/52013479/191824787-f613d17f-e37e-43a0-8ff7-d755a2261386.png)

**Comment by SHHSSH** [comment]
Good luck with it if so, if I recall Sideberry was pretty lightweight and I enjoyed it also.

**Comment by Elaws** [comment]
11 days after installing Sidebery : it's a blast to use ! Fast, clean UI, no bugs yet. Perfect STG + TST replacement (for my usecase at least).

**Comment by SHHSSH** [comment]
> 11 days after installing Sidebery : it's a blast to use ! Fast, clean UI, no bugs yet. The perfect STG + TST replacement (for my usecase at least) I should've installed years ago.

Yeah I can imagine it'd be much more lightweight than TST, I just love my customised TST :D
